tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -101,6 +74,29 @@ def _get_test_source_directory() -> Path:
             path = path.parent
         raise Exception("Can't find the test root directory (tests_e2e)")
 
+    def _setup(self) -> None:
+        """
+        Prepares the test scenario for execution
+        """
+        self._log.info(f"Test Node: {self._context.vm_name}")
+        self._log.info(f"Resource Group: {self._context.resource_group_name}")
+        self._log.info(f"Working directory: {self._context.working_directory}...")
+
+        if self._context.working_directory.exists():
+            self._log.info(f"Removing existing working directory: {self._context.working_directory}...")
+            try:
+                rmtree(self._context.working_directory.as_posix())
+            except Exception as exception:
+                self._log.warning(f"Failed to remove the working directory: {exception}")
+        self._context.working_directory.mkdir()
```

Comment:
continue with warning. this is mainly for the dev scenario. the automation run works on fresh vms

Reply:
I was reading mkkdir() raise exception if directory exists. So, in dev scenario it won't work then

Reply:
My point is we could raise the exception if we failed to remove because anyway in next step mkdir throws exception if folder exist.

Reply:
yes, it will fail and there will be a warning with the info about the delete failure

the dev will get an error, see the warning in the log and fix the issue

Reply:
> My point is we could raise the exception if we failed to remove because anyway in next step mkdir throws exception if folder exist.

Same difference, with the possibility that the working directory goes away in-between the 2 calls and there is no error